includes/class-wc-payments-express-checkout-button-display-handler.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Class WC_Payments_Express_Checkout_Button_Display_Handler
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class WC_Payments_Express_Checkout_Button_Display_Handler
+ */
+class WC_Payments_Express_Checkout_Button_Display_Handler {
+
+	/**
+	 * WC_Payment_Gateway_WCPay instance.
+	 *
+	 * @var WC_Payment_Gateway_WCPay
+	 */
+	private $gateway;
+
+	/**
+	 * Instance of WC_Payments_Payment_Request_Button_Handler, created in init function
+	 *
+	 * @var WC_Payments_Payment_Request_Button_Handler
+	 */
+	private $payment_request_button_handler;
+
+	/**
+	 * Instance of WC_Payments_Platform_Checkout_Button_Handler, created in init function
+	 *
+	 * @var WC_Payments_Platform_Checkout_Button_Handler
+	 */
+	private $platform_checkout_button_handler;
+
+	/**
+	 * Initialize class actions.
+	 *
+	 * @param WC_Payment_Gateway_WCPay                     $gateway WCPay gateway.
+	 * @param WC_Payments_Payment_Request_Button_Handler   $payment_request_button_handler Payment request button handler.
+	 * @param WC_Payments_Platform_Checkout_Button_Handler $platform_checkout_button_handler Platform checkout button handler.
+	 */
+	public function __construct( WC_Payment_Gateway_WCPay $gateway, WC_Payments_Payment_Request_Button_Handler $payment_request_button_handler, WC_Payments_Platform_Checkout_Button_Handler $platform_checkout_button_handler ) {
+		$this->gateway                          = $gateway;
+		$this->payment_request_button_handler   = $payment_request_button_handler;
+		$this->platform_checkout_button_handler = $platform_checkout_button_handler;
+
+		$this->platform_checkout_button_handler->init();
+		$this->payment_request_button_handler->init();
+
+		$is_woopay_express_checkout_enabled = WC_Payments_Features::is_woopay_express_checkout_enabled();
+
+		if ( $is_woopay_express_checkout_enabled || $this->gateway->get_option( 'payment_request' ) ) {
```

Comment:
```suggestion
		if ( $is_woopay_express_checkout_enabled || 'yes' === $this->gateway->get_option( 'payment_request' ) ) {
```
AFAIA, the value of `$this->gateway->get_option( 'payment_request' )` can be either `yes` or `no`. Presumably, we are checking for more than the presence of this setting, but rather that it has also been enabled as well. The same goes for the identical check below on line 60, but correct me, if I'm misunderstanding something here!

Reply:
>the value of $this->gateway->get_option( 'payment_request' ) can be either yes or no

That's correct! Nice catch.